src/main/java/Model/Game.java

```diff
@@ -1,269 +1,383 @@
+        colouredPiecesArray.add(new Piece(isWhite, PieceType.Rook));
+        colouredPiecesArray.add(new Piece(isWhite, PieceType.Knight));
+        colouredPiecesArray.add(new Piece(isWhite, PieceType.Bishop));
+        colouredPiecesArray.add(new Piece(isWhite, PieceType.Queen));
+        colouredPiecesArray.add(new Piece(isWhite, PieceType.King));
+        colouredPiecesArray.add(new Piece(isWhite, PieceType.Bishop));
+        colouredPiecesArray.add(new Piece(isWhite, PieceType.Knight));
+        colouredPiecesArray.add(new Piece(isWhite, PieceType.Rook));
 
         for(int i = 8; i < 16; i++) {
-            addPiece(new Position(i-8, rank), pieces.get(i));
+            board.addPiece(new Position(i-8, rank), colouredPiecesArray.get(i));
         }
     }
 
-    /**
-     * basically create a copy of boardSquares to help create the first FEN
-     * 
-     * @return 2d array of the first fen position
-     */
-    private Piece[][] exportBoardPositionForFEN() {
-        Piece[][] piece2DArray = new Piece[8][8];
-        for (int rank = 0; rank < 8; rank++) {
-            piece2DArray[rank] = boardSquares.getRankPiece(rank);
-        }
-        return piece2DArray;
-    }
 
-    /**
-     * Add the piece to the specific Square on the file and rank provided
-     * 
-     * @param position
-     * @param piece
-     */
-    private void addPiece(Position position, Piece piece) {
-        boardSquares.addPiece(position, piece);
-    }
 
-    /**
-     * remove the piece on the specific position
-     * 
-     * @param position
-     */
-    private void removePiece(Position position) {
-        boardSquares.removePiece(position);
-    }
 
-    /**
-     * Change the whitesTurn
-     */
-    private void nextTurn() {
-        whitesTurn = !whitesTurn;
-        fullmove+=whitesTurn?1:0;
-    }
+
+
+
+
+
+
+
+
+
 
     /**
      * Move the piece from the previousPosition to the newPosition.
-     * Also checking for en Passant and promotion. Then update the view
-     * 
-     * @param previousPosition
-     * @param newPosition
+     * Also checking for en Passant and promotion. Then update the view and endTurn
+     *
+     * @param previousPosition current position of the piece
+     * @param newPosition where the piece will be moving to
      */
-    public void movePieces(Position previousPosition, Position newPosition) {
-        Piece piece = boardSquares.getPiece(previousPosition);
-
-        // if en Passant occured, return as the checkForEnPassant function did everything necessary
-        if(checkForEnPassant(previousPosition, newPosition, piece)) {
+    public void moveAMove(Position previousPosition, Position newPosition) {
+        if (isNull(getPiece(previousPosition)) || getPiece(previousPosition).getIsWhite()!=whitesTurn){
             return;
         }
 
+        if (getLegalMoves(previousPosition).stream().noneMatch(position -> position.getFile()== newPosition.getFile() && position.getRank()== newPosition.getRank())){
```

Comment:
this needs a comment. assuming you checking that the new move is not part of the legal moves from the last turn?

Reply:
This is making sure that `newPosition `is a Legal move from the `previousPosition`